app/lib/feed_manager.rb

```diff
@@ -351,10 +351,16 @@ def blocks_or_mutes?(receiver_id, account_ids, context)
   # @param [Integer] receiver_id
   # @param [Hash] crutches
   # @return [Boolean]
-  def filter_from_home?(status, receiver_id, crutches)
+  def filter_from_home?(status, receiver_id, crutches, timeline_type = :home)
     return false if receiver_id == status.account_id
     return true  if status.reply? && (status.in_reply_to_id.nil? || status.in_reply_to_account_id.nil?)
-    return true  if crutches[:languages][status.account_id].present? && status.language.present? && !crutches[:languages][status.account_id].include?(status.language)
+    unless timeline_type == :list
+      # find all exclusive lists
+      lists = List.where(account_id: receiver_id, exclusive: true)
+      # is there a list the receiver owns with this account on it? if so, return true
+      return true if ListAccount.where(list: lists, account_id: status.account_id).exists?
```

Comment:
Since this runs for all `:home` updates, it needs to work with `crutches` or it will cause a major slowdown during home feed regeneration due to an N+1 query.

Reply:
@dariusk

Reply:
I've already got a PR on Darius' fork (https://github.com/dariusk/mastodon/pull/10) that resolves this + merge conflicts + linting errors, and he's already aware of it. I expect it'll get merged in soon!